imports/izar_protocol_v3.leo

```diff
@@ -0,0 +1,270 @@
+
+    struct Nonce {
+        chain_id: u32,
+        nonce: u128,
+    }
+
+    inline contract_owner() -> address {
+        return aleo1yu6ssnehj4plgznatenfut3hkrfus26kqd226k99wfx79jqmsuxqfernns;
+    }
+
+    inline unvalid_contract() -> address {
+        return aleo1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq3ljyzc;
+    }
+
+    /* config setup */
+    transition set_epoch (epoch: u32) {
+        assert_eq(self.caller, contract_owner());
+        return then finalize (epoch);
+    }
+
+    finalize set_epoch (epoch: u32) {
+        assert(epoch > Mapping::get_or_use(izar_config, 0u8, 0u32));
+        Mapping::set(izar_config, 0u8, epoch);
+    }
+
+    transition update_keeper (keepers: Keepers20) {
+        assert_eq(self.caller, contract_owner());
+        return then finalize (keepers);
+    }
+
+    finalize update_keeper (keepers: Keepers20) {
+        let accpeted: u32 = 0u32;
+
+        if keepers.k0 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k1 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k2 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k3 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k4 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k5 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k6 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k7 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k8 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k9 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k10 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k11 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k12 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k13 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k14 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k15 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k16 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k17 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k18 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        if keepers.k19 != unvalid_contract() {
+            accpeted += 1u32;
+        }
+
+        Mapping::set(izar_keeper, true, keepers);
+        Mapping::set(izar_config, 1u8, accpeted);
+    }
+
+    transition verify (from_chain_id: u32, signatures: Signatures20, keepers: Keepers20, payload: IzarRecvMsg) {
```

Comment:
Should put `from_chain_id` into `IzarRecvMsg`

Reply:
Fixed